plugins/modules/idrac_server_config_profile.py

```diff
@@ -755,83 +752,178 @@ def preview_scp_redfish(module, idrac, http_share, import_job_wait=False):
-                     "username": module.params.get("share_user"), "password": module.params.get("share_password")}
-            if http_share == "HTTPS":
-                share["ignore_certificate_warning"] = IGNORE_WARNING[module.params["ignore_certificate_warning"]]
-        else:
-            share, scp_file_name_format = get_scp_share_details(module)
-            share["file_name"] = module.params.get("scp_file")
-        buffer_text = None
+        share, _scp_file_name_format = get_scp_share_details(module)
+        share["file_name"] = module.params.get("scp_file")
+        buffer_text = get_buffer_text(module, share)
         share_dict = share
         if share["share_type"] == "LOCAL":
-            file_path = "{0}{1}{2}".format(share["share_name"], os.sep, share["file_name"])
-            if not exists(file_path):
-                module.fail_json(msg=INVALID_FILE)
-            with open(file_path, "r") as file_obj:
-                buffer_text = file_obj.read()
             share_dict = {}
         module.params["job_wait"] = job_wait
-        scp_response = idrac.import_scp_share(shutdown_type=module.params["shutdown_type"],
-                                              host_powerstate=module.params["end_host_power_state"],
-                                              job_wait=module.params["job_wait"],
-                                              target=scp_targets,
-                                              import_buffer=buffer_text, share=share_dict, )
-        job_id = scp_response.headers["Location"].split("/")[-1]
-        if module.params["job_wait"]:
-            job_failed, msg, job_dict, wait_time = idrac_redfish_job_tracking(
-                idrac, iDRAC_JOB_URI.format(job_id=job_id))
-            scp_response = job_dict
+        idrac_import_scp_params = {
+            "import_buffer": buffer_text, "target": scp_targets, "share": share_dict, "job_wait": module.params["job_wait"],
+            "host_powerstate": module.params["end_host_power_state"], "shutdown_type": module.params["shutdown_type"]
+        }
+        scp_response = idrac_import_scp_share(module, idrac, idrac_import_scp_params)
     else:
         scp_response = idrac.import_scp(import_buffer=import_buffer, target=scp_targets, job_wait=module.params["job_wait"])
     scp_response = response_format_change(scp_response, module.params, share.get("file_name"))
-    if isinstance(scp_response, dict) and scp_response.get("TaskStatus") == "Critical":
-        module.fail_json(msg="Failed to {0} scp.".format(command), scp_status=scp_response)
+    exit_on_failure(module, scp_response, command)
     return scp_response
 
 
+def idrac_import_scp_share(module, idrac, job_params):
+    scp_response = idrac.import_scp_share(**job_params)
+    job_id = scp_response.headers["Location"].split("/")[-1]
+    if module.params["job_wait"]:
+        job_failed, _msg, job_dict, _wait_time = idrac_redfish_job_tracking(
```

Comment:
I can see this snippet is doing job_tracking but function name is not matching with what function is doing, can you change it ?

Reply:
this function performs import it's not specific to job tracking

Reply:
If we pass the resp directly to this function we won't have to write another two function for export and preview, we can simply pass the resp and rest will same. 

But currently it's dependent on idrac.import_scp_share, I think it's better we pass the resp and use this.

Reply:
as discussed, as part of this PR changes, job tracking is not re-used across workflows.